pydruid/db/api.py

```diff
@@ -39,6 +40,14 @@ def connect(
     """
     context = context or {}
 
+    if header is not None:
+        warnings.warn(
+            "The `header` parameter is deprecated and will be removed in a future release",  # noqa: E501
+            DeprecationWarning,
+        )
+        if header is not True:
+            raise ValueError("Disabling the column header is no longer supported.")
```

Comment:
Since we want to maintain compatibility with upstream pydruid, and upstream pydruid still supports header, and doesn't seem to have any plans to deprecate it, I would suggest we should not deprecate or remove it.

Thus, I wouldn't warn if header is specified as True. I would warn if it is specified as False, rather than error out, warning that it will be ignored in this version of the lib.

Reply:
I'll refactor the PR to only use `arrayLines` when `header=True`

Reply:
That will complexify the code. I like how arraylines simplifies things.

Reply:
Ohh I misread your previous suggestion :man_facepalming: . I'll change the deprecation message to only apply when `header=False`